src/rules.jl

```diff
@@ -162,19 +167,19 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
 #--------#
 
 @define_diffrule SpecialFunctions.besselj(ν, x)   =
-    :NaN, :(  (SpecialFunctions.besselj($ν - 1, $x) - SpecialFunctions.besselj($ν + 1, $x)) / 2  )
+    :(oftype(float($ν), NaN)), :(  (SpecialFunctions.besselj($ν - 1, $x) - SpecialFunctions.besselj($ν + 1, $x)) / 2  )
```

Comment:
Is this needed? If this partial is included in any computations it will break them anyway.

Reply:
Up to you tbh. I like the consistency, but I'm also fine with not doing it (just need to filter these during testing).

Reply:
Hm I think I prefer if we keep `NaN` if there are no clear reasons to change it. `NaN` seems a bit simpler and in DiffRules usually indicates if a derivative does not exist or is not implemented. There are also some other rules that still return `:NaN` in this PR. My suggestion would be to keep `:NaN` and ensure type stability and avoid promotions in cases where only sometimes `NaN` is returned (e.g., `isinteger(x) ? NaN : 1`).

Reply:
Aight, will do :+1: I'll just filter them out from the tests then.